### PR TITLE
AWS Secrets Manager support

### PR DESCRIPTION
**Pull request by kichristensen**
Add support for AWS Secrets Manager

related to https://github.com/getporter/porter/issues/3171

### CODE REVIEW

File: .github/workflows/build.yaml [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+env:
+  # Make mage super verbose ALL THE TIME
+  MAGEFILE_VERBOSE: true
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
```
Comment (dgannon991) [inline .github/workflows/build.yaml:14]: Tofu mixin uses checkout@v4, not sure of the differences?

File: .github/workflows/build.yaml [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+        with:
+          fetch-depth: 0
+      - uses: actions/setup-go@v3
```
Comment (dgannon991) [inline .github/workflows/build.yaml:17]: Tofu uses setup-go@v5

File: .vscode/scripts/runporterinbackground.sh
Comment (dgannon991) [inline .vscode/scripts/runporterinbackground.sh:1]: Attempting to run this script on Fedora results in:

```
./runporterinbackground.sh: line 3: tempfile: command not found
./runporterinbackground.sh: line 5: ${OUTPUT}: ambiguous redirect
dlv server not running after 3 seconds
```

And the script hangs.

File: .vscode/launch.json [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+      "dlvLoadConfig":{
+        "followPointers": true,
+        "maxVariableRecurse": 1,
+        "maxStringLen": 128,
+        "maxArrayValues": 64,
+        "maxStructFields": -1
+      }
+    },
+    {
+      "preLaunchTask": "SetupPluginAndRunPorter",
```
Comment (dgannon991) [inline .vscode/launch.json:31]: Attempting to launch this fails saying it cannot find the task `SetupPluginAndRunPorter`

File: CONTRIBUTING.md
Comment (dgannon991) [inline CONTRIBUTING.md:1]: Would it be possible to add a section here on how to use the debugging scripts and the launch JSON? It's probably just me doing something silly, but I've been unable to launch either scenario.

**Review by dgannon991 — COMMENTED** [review]
Just some general thoughts, most are probably my error :D I've never done a review here before, so apologies if I've got this slightly wrong!

File: .github/workflows/build.yaml (hunk 1 above)
Comment (kichristensen) [reply .github/workflows/build.yaml:14]: Shouldn't make a big difference, but should be updated nontheless. I have updated it now

File: .github/workflows/build.yaml (hunk 2 above)
Comment (kichristensen) [reply .github/workflows/build.yaml:17]: Shouldn't make a big difference, but should be updated nontheless. I have updated it now

File: CONTRIBUTING.md
Comment (kichristensen) [reply CONTRIBUTING.md:1]: I  will take a look at it, honestly it is primarily copy paste from other plugins :) But it does make sense to add

File: .vscode/scripts/runporterinbackground.sh
Comment (kichristensen) [reply .vscode/scripts/runporterinbackground.sh:1]: I have remove the whole task, and refer to the official Porter plugin debugging guide instead.

File: .vscode/launch.json (hunk 3 above)
Comment (kichristensen) [reply .vscode/launch.json]: I have remove the whole task, and refer to the official Porter plugin debugging guide instead.

File: CONTRIBUTING.md
Comment (kichristensen) [reply CONTRIBUTING.md:1]: I have remove the whole task, and refer to the official Porter plugin debugging guide instead.

**Review by dgannon991 — APPROVED** [review]

**Pull request merged**